cluster-autoscaler/core/scaledown/eligibility/eligibility_test.go

```diff
@@ -0,0 +1,137 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package eligibility
+
+import (
+	"strconv"
+	"testing"
+	"time"
+
+	"k8s.io/autoscaler/cluster-autoscaler/cloudprovider"
+	testprovider "k8s.io/autoscaler/cluster-autoscaler/cloudprovider/test"
+	"k8s.io/autoscaler/cluster-autoscaler/config"
+	"k8s.io/autoscaler/cluster-autoscaler/context"
+	"k8s.io/autoscaler/cluster-autoscaler/core/scaledown/unremovable"
+	. "k8s.io/autoscaler/cluster-autoscaler/core/test"
+	"k8s.io/autoscaler/cluster-autoscaler/simulator"
+	"k8s.io/autoscaler/cluster-autoscaler/utils/deletetaint"
+	. "k8s.io/autoscaler/cluster-autoscaler/utils/test"
+
+	"github.com/stretchr/testify/assert"
+	apiv1 "k8s.io/api/core/v1"
+	"k8s.io/client-go/kubernetes/fake"
+)
+
+func TestFilterOutUnremovable(t *testing.T) {
+	now := time.Now()
+
+	regularNode := BuildTestNode("regular", 1000, 10)
+	SetNodeReadyState(regularNode, true, time.Time{})
+
+	justDeletedNode := BuildTestNode("justDeleted", 1000, 10)
+	justDeletedNode.Spec.Taints = []apiv1.Taint{{Key: deletetaint.ToBeDeletedTaint, Value: strconv.FormatInt(now.Unix()-30, 10)}}
+	SetNodeReadyState(justDeletedNode, true, time.Time{})
+
+	tooOldDeletedNode := BuildTestNode("tooOldDeleted", 1000, 10)
+	tooOldDeletedNode.Spec.Taints = []apiv1.Taint{{Key: deletetaint.ToBeDeletedTaint, Value: strconv.FormatInt(now.Unix()-301, 10)}}
+	SetNodeReadyState(tooOldDeletedNode, true, time.Time{})
+
+	noScaleDownNode := BuildTestNode("noScaleDown", 1000, 10)
+	noScaleDownNode.Annotations = map[string]string{ScaleDownDisabledKey: "true"}
+	SetNodeReadyState(noScaleDownNode, true, time.Time{})
+
+	bigPod := BuildTestPod("bigPod", 600, 0)
+	bigPod.Spec.NodeName = "regular"
+
+	smallPod := BuildTestPod("smallPod", 100, 0)
+	smallPod.Spec.NodeName = "regular"
+
+	testCases := []struct {
+		desc  string
+		nodes []*apiv1.Node
+		pods  []*apiv1.Pod
+		want  []string
+	}{
+		{
+			desc:  "regular node stays",
+			nodes: []*apiv1.Node{regularNode},
+			want:  []string{"regular"},
+		},
+		{
+			desc:  "recently deleted node is filtered out",
+			nodes: []*apiv1.Node{regularNode, justDeletedNode},
+			want:  []string{"regular"},
+		},
+		{
+			desc:  "deleted long time ago stays",
```

Comment:
nit: I don't like this case in this specific test. I'd argue it mainly tests the specific implementation of actuation.IsNodeBeingDeleted() more than it tests logic in this package.